### DISCUSSION TITLE
Disable click on Navigation Menu Trigger

### DISCUSSION THREAD

**Discussion by ayaniv** [radix-ui/primitives#2404] [Ideas]
The Radix Navigation Menu is excellent, and I particularly appreciate its accessibility features. However, there is a potential issue that may arise due to its dual functionality of responding to both hover and click actions on the NavigationMenuTrigger. This could lead to a confusing bug where clicking on the trigger inadvertently closes the menu instead of displaying its content.

For instance, you can observe this behavior by attempting to open the dropdown by clicking on the trigger in the original example from the documentation:

[Original Example Link](https://codesandbox.io/p/sandbox/8vvktr?file=%2FApp.jsx%3A1%2C1)

To address this issue, I'd like to propose the following modifications, which prevent mouse clicks on the trigger while still allowing keyboard actions to function. You can review the code changes in the following link:

[Modified Code Link](https://codesandbox.io/s/cool-bush-8zvsnz?file=/src/App.js:2535-2560)

In these changes, I've introduced a new component named StyledTriggerDisableClick, as outlined in lines 98-131. Then, I've substituted this component for StyledTrigger within StyledTriggerWithCaret, as seen in lines 135 and 138. The core concept behind this adjustment is to set a flag to true when hovering over the trigger element and to check this flag when clicking the trigger to prevent mouse clicks from closing the menu inadvertently.

I'm sharing this workaround for the benefit of future users who may want to block mouse click events on the Radix Navigation Menu Trigger. Additionally, I'm submitting this as a feature request for the next version, in hopes of enabling developers to achieve this behavior with a simple prop on the trigger.

Your feedback and comments on these suggestions are highly appreciated.

**Comment by ayaniv** [comment]
The suggested solution above is not so good as it blocks touch devices from working properly.

**Comment by martijnmichel** [comment]
You could try this: 
```ts
<NavigationMenu.Trigger
            onClick={(e) => {
              console.log(e);
              if (
                "pointerType" in e.nativeEvent &&
                e.nativeEvent.pointerType === "touch"
              ) {
               console.log("touch event, respond normally");
              } else {
                e.preventDefault();
              }
            }}
          >
          ```
